### PR TITLE
Gang is telegraphed but much more delayed

### PR DESCRIPTION
**Pull request by Cyberboss**
:cl:
balance: Gang has a new intro
/:cl:

Implementation of @KorPhaeron's idea from here: https://github.com/tgstation/tgstation/pull/27145#issuecomment-301114352

> Something like
> 
> Minute 0: Security explodes
> 
> Minute 3: You've decided to take charge of the situation and form a gang. X and Y are your ganghead friends [syndicate company] has offered an uplink to help you take over, it will be delivered in five minutes.
> 
> Minute 8: You get your uplink and hopefully made up a plan in that time.
> 
> Delays might be a bit long, just throwing random numbers out but it might stop the stetchkin rush we'd get otherwise.
>

### CODE REVIEW

**Comment by TehZombehz** [comment]
Why has security not exploded BEFORE minute 0?

File: code/game/gamemodes/gang/gang.dm [hunk 1]
```diff
@@ -69,13 +80,33 @@ GLOBAL_LIST_INIT(gang_colors_pool, list("red","orange","yellow","green","blue","
...
+	return 1
 
-/datum/game_mode/gang/post_setup()
+/datum/game_mode/gang/send_intercept()
 	set waitfor = FALSE
-	..()
-	sleep(rand(10,100))
+	..(FALSE, FALSE)
+
+	priority_announce("Incoming Bluespace Artillery Strike", "Target location: Security department. Evacuate immediately", 'sound/machines/Alarm.ogg')
```
Comment (KorPhaeron) [inline code/game/gamemodes/gang/gang.dm]: Syndicate doesn't have bluespace artillery!

Comment (coiax) [inline code/game/gamemodes/gang/gang.dm]: Who's shooting the station?

File: code/game/gamemodes/gang/gang.dm [hunk 2]
```diff
@@ -69,13 +80,33 @@ GLOBAL_LIST_INIT(gang_colors_pool, list("red","orange","yellow","green","blue","
...
+
+	sleep(100)
+
+	explosion(pick(get_area_turfs(target_armory)), 7, 14, 28, 30, TRUE, TRUE)
+
+	sleep(1200)
+
+	for(var/datum/gang/G in gangs)
+		for(var/datum/mind/boss_mind in G.bosses)
+			to_chat(boss_mind.current, "<span class='danger'>You've decided to take charge of the situation and form a gang. X and Y are your ganghead friends [syndicate company] has offered an uplink to help you take over, it will be delivered in five minutes.</span>")
```
Comment (KorPhaeron) [inline code/game/gamemodes/gang/gang.dm]: [syndicate company] should pick from a list of the factions, or just the name of the gang has offered or something

X and Y should be the names of the other gang heads

File: code/game/gamemodes/gang/gang.dm [hunk 3]
```diff
@@ -69,13 +80,33 @@ GLOBAL_LIST_INIT(gang_colors_pool, list("red","orange","yellow","green","blue","
 	if(gangs.len < 2) //Need at least two gangs
 		return 0
 
-	return 1
+	//turn off sec and captain
+	SSjob.DisableJob(/datum/job/captain)
```
Comment (coiax) [inline code/game/gamemodes/gang/gang.dm]: Why no captain? The Hop will just take command as captain.

File: code/game/gamemodes/gang/gang.dm [hunk 4]
```diff
@@ -28,16 +28,27 @@ GLOBAL_LIST_INIT(gang_colors_pool, list("red","orange","yellow","green","blue","
...
 
+	var/area/ai_monitored/security/armory/target_armory
+
 ///////////////////////////////////////////////////////////////////////////////
 //Gets the round setup, cancelling if there's not enough players at the start//
 ///////////////////////////////////////////////////////////////////////////////
 /datum/game_mode/gang/pre_setup()
+	for(var/area/ai_monitored/security/armory/A in GLOB.sortedAreas)
+		if(A.z == ZLEVEL_STATION)
+			target_armory = A
```
Comment (KorPhaeron) [inline code/game/gamemodes/gang/gang.dm]: This probably needs to trash the security equipment room as well

**Comment by PKPenguin321** [comment]
This with the delays and no infinite shuttle recalls will probably make rushing the shuttle break the mode

File: code/game/gamemodes/gang/gang.dm (hunk 3 above)
Comment (KorPhaeron) [inline code/game/gamemodes/gang/gang.dm]: Captain leading the vigilantes might be interesting I guess

**Comment by KorPhaeron** [comment]
@PKPenguin321 

You can't even call the shuttle until 22 minutes in

**Comment by PKPenguin321** [comment]
Oh

File: code/game/gamemodes/gang/gang.dm (hunk 4 above)
Comment (KorPhaeron) [inline code/game/gamemodes/gang/gang.dm]: @Robustin do you think stopping cargo from buying implants is a good idea or should the vigilantes be able to form their own new security

File: code/game/gamemodes/gang/gang.dm (hunk 3 above)
Comment (Cyberboss) [inline code/game/gamemodes/gang/gang.dm]: I just thought might as well kill off the last guy with an implant

File: code/game/gamemodes/gang/gang.dm (hunk 4 above)
Comment (MMMiracles) [inline code/game/gamemodes/gang/gang.dm]: I don't personally see a reason in out-right removing any chance of a loyalist group to compete against gangs. If the armory and equipment is getting bombarded round-start, it becomes a lot harder to mass capture/implant.

File: code/game/gamemodes/gang/gang.dm (hunk 1 above)
Comment (Cyberboss) [inline code/game/gamemodes/gang/gang.dm]: Ideas?

File: code/game/gamemodes/gang/gang.dm [hunk 5]
```diff
@@ -69,20 +94,55 @@ GLOBAL_LIST_INIT(gang_colors_pool, list("red","orange","yellow","green","blue","
...
+		var/list/other_bosses = bosses.Copy() - boss_mind
+		var/msg = "<span class='danger'>You've decided to take charge of the situation and form a gang."
+		for(var/J in 1 to other_bosses.len)
+			var/datum/mind/other_mind = other_bosses[J]
+			if(J == other_bosses.len)
+				msg += " and"
+			msg += " J"
+			if(J != other_bosses.len)
+				msg += ","
+		to_chat(boss_mind.current, "[msg] are your ganghead rivals. [syndicate_name()] has offered an uplink to help you take over, it will be delivered in five minutes.</span>")
```
Comment (ChangelingRain) [inline code/game/gamemodes/gang/gang.dm]: the syndicate name should include the gang's name, so you get, like, cybersun supporting the cyber gang, etc

File: code/game/gamemodes/gang/gang.dm (hunk 3 above)
Comment (Iamgoofball) [inline code/game/gamemodes/gang/gang.dm]: captain is just assistant with all access roundstart so i dont think he should be disabled

the non-gang group needs some form of leader after all

File: code/game/gamemodes/gang/gang.dm (hunk 1 above)
Comment (Robustin) [inline code/game/gamemodes/gang/gang.dm]: Turrurism

File: tools/tgstation-server-3/TGServerService/bin/Debug/TGServerService.InstallLog [hunk 6]
```diff
@@ -0,0 +1,13 @@
+﻿Installing assembly 'D:\tgstation\tools\tgstation-server-3\TGServerService\bin\Debug\TGServerService.exe'.
```
Comment (Iamgoofball) [inline tools/tgstation-server-3/TGServerService/bin/Debug/TGServerService.InstallLog]: >when you add your server tools by accident

Comment (Cyberboss) [inline tools/tgstation-server-3/TGServerService/bin/Debug/TGServerService.InstallLog]: rekt

File: code/game/gamemodes/gang/gang.dm [hunk 7]
```diff
@@ -69,18 +93,54 @@ GLOBAL_LIST_INIT(gang_colors_pool, list("red","orange","yellow","green","blue","
 	if(gangs.len < 2) //Need at least two gangs
 		return 0
 
-	return 1
+	//turn off sec and captain
+	SSjob.DisableJob(/datum/job/captain)
+	SSjob.DisableJob(/datum/job/hos)
+	SSjob.DisableJob(/datum/job/warden)
+	SSjob.DisableJob(/datum/job/detective)
```
Comment (Iamgoofball) [inline code/game/gamemodes/gang/gang.dm]: Detective isn't security anymore

File: code/game/gamemodes/gang/gang.dm [hunk 8]
```diff
@@ -69,18 +93,54 @@ GLOBAL_LIST_INIT(gang_colors_pool, list("red","orange","yellow","green","blue","
...
+				if(J == other_bosses.len)
+					msg += " and"
+				msg += " [other_mind.current]"
+				if(J != other_bosses.len)
+					msg += ","
+			to_chat(boss_mind.current, "[msg] are your ganghead rivals. [syndicate_name()] has offered an uplink to help you take over, it will be delivered in five minutes.</span>")
 			forge_gang_objectives(boss_mind)
 			greet_gang(boss_mind)
+	
+	sleep(3000)
```
Comment (Iamgoofball) [inline code/game/gamemodes/gang/gang.dm]: addtimer

File: code/game/gamemodes/gang/gang.dm [hunk 9]
```diff
@@ -69,18 +93,54 @@ GLOBAL_LIST_INIT(gang_colors_pool, list("red","orange","yellow","green","blue","
...
+	..(FALSE, FALSE)
+
+	priority_announce("Incoming Bluespace Artillery Strike", "Target location: Security department. Evacuate immediately", 'sound/machines/Alarm.ogg')
+
+	sleep(100)
+
+	explosion(target_armory, 7, 14, 28, 30, TRUE, TRUE)
+	explosion(target_equipment, 7, 14, 28, 30, TRUE, TRUE)
+
+	sleep(1200)
```
Comment (Iamgoofball) [inline code/game/gamemodes/gang/gang.dm]: addtimer

File: code/game/gamemodes/gang/gang.dm [hunk 10]
```diff
@@ -69,18 +93,54 @@ GLOBAL_LIST_INIT(gang_colors_pool, list("red","orange","yellow","green","blue","
...
-/datum/game_mode/gang/post_setup()
+/datum/game_mode/gang/send_intercept()
 	set waitfor = FALSE
-	..()
-	sleep(rand(10,100))
+	..(FALSE, FALSE)
+
+	priority_announce("Incoming Bluespace Artillery Strike", "Target location: Security department. Evacuate immediately", 'sound/machines/Alarm.ogg')
+
+	sleep(100)
```
Comment (Iamgoofball) [inline code/game/gamemodes/gang/gang.dm]: addtimer

**Comment by Cyberboss** [comment]
It's a detached proc on game_mode so making it addtimer instead of sleeps is pointless

**Comment by Robustin** [comment]
This feels a little too half-assed.

1) Cargo gun orders still need to be disabled

2) I had some lore ideas but "Hey its time to take charge of the situation! Also we happen to suddenly have the ability to mind-control people with pens and summon guns out of handheld devices!" is a pretty shitty one. 

3) The delays don't feel good at all. I really don't look forward to what happens when everyone realizes there's no security and everyone will be invariably turned into an antag, but there's about 8 or so minutes where all you can do is loot everything and riot until you get pen'd. 

4) I didn't want to attempt this until I had a purpose for non-gangsters besides "beg on radio for someone to convert you". Because thats literally all thats going to get accomplished: Looting, Fighting, and then a few minutes later "HEY SOMEONE RECRUIT ME I WANNA SHOOT SHIT".

**Comment by Iamgoofball** [comment]
1. That's easy to do.
2. l o r e
3. Actually, I'd say that's good. People don't immediately go from disaster to organized, having that time of folks scrambling around like headless chickens will help with the feel.
4. let's be fair here this is all conversion gamemodes

**Comment by Iatots** [comment]
Make the initial arty strike be a gang drive-by, and give spawning gang heads a message on how they lost faith in nanotrasen to uphold the law and now have to defend their turf themselves.

**Comment by Iamgoofball** [comment]
the ballas thought that security was part of grove street

there you go, security getting exploded lore

**Comment by Iamgoofball** [comment]
https://github.com/Cyberboss/tgstation/pull/4

added gang lore

@Cyberboss pls merge

**Comment by Cyberboss** [comment]
https://i.imgflip.com/1ox3xy.jpg

**Comment by RemieRichards** [comment]
That lore isn't getting in, I hope you know that.

**Comment by Iamgoofball** [comment]
fuckin grove motherfucker

**Comment by RemieRichards** [comment]
It's also -just not good lore- so, it will be changed at some point, but the toned down version is "ok" for now.

**Comment by KorPhaeron** [comment]
I have even sillier ideas to build on this

https://tgstation13.org/phpBB/viewtopic.php?f=9&t=11130&view=unread#unread

File: code/game/gamemodes/gang/gang.dm (hunk 4 above)
Comment (octareenroon91) [inline code/game/gamemodes/gang/gang.dm]: Implants matter because of implant-keyed firing pins.

File: code/game/gamemodes/gang/gang.dm (hunk 3 above)
Comment (coiax) [inline code/game/gamemodes/gang/gang.dm]: Maybe take away his roundstart implant?

**Comment by Iamgoofball** [comment]
@KorPhaeron can you get in irc fam

**Comment by Robustin** [comment]
@Iamgoofball 

Except they won't be scrambling around like headless chicken they'll be playing like they just rolled antag (back when they started human). Steal supplies, fortify, and start to stack the odds in your favor (make bombs, subvert AI, start ordering weapons, make chem nades, kudzu, whatever. Nobody will be confused except new players - the only other "confusion" will be ahelps asking if it was acceptable for the CE to RCD into medical supplies and steal everything. 

Nothing that enhances the round will happen when everyone is playing like its the start of a purge round. All it means is the biggest powergamers will be ready to murder half the station before anyone even gets true antag status. There's no good reason to have that kind of delay. 

I'm trying to streamline the mode but having almost 10 minutes "waiting for antag status that everyone knows is going to happen" is just bad.

File: code/game/gamemodes/gang/gang.dm (hunk 3 above)
Comment (Dorsisdwarf) [inline code/game/gamemodes/gang/gang.dm]: I like the idea of just the captain being the Only Sane Man trying to fend off the rabid murderers

**Comment by Iamgoofball** [comment]
@Robustin You may be right, and I may be right. We should testmerge this to see who's right, instead of arguing over what MIGHT happen.

**Comment by KorPhaeron** [comment]
Nothing wrong with the crew rioting in a mode that is team death match at its most basic

**Comment by KorPhaeron** [comment]
I guess we could avoid the delay all together if gang bosses can create hideouts though

**Comment by octareenroon91** [comment]
What, like with a child of the bluespace shelter capsule?

**Comment by Robustin** [comment]
@KorPhaeron 

Yea and if the most is going to be TDM, lets get down to business and not have 8 minutes of filler-rioting.

**Comment by kevinz000** [comment]
> @Robustin thinks tdm is good

lulz

**Comment by Robustin** [comment]
>Looks at 200 people playing on CM 

Yea I gotta be the only one who likes the occasional deathmatch.

File: code/game/gamemodes/gang/gang.dm [hunk 11]
```diff
@@ -69,18 +93,54 @@ GLOBAL_LIST_INIT(gang_colors_pool, list("red","orange","yellow","green","blue","
...
+	return 1
 
-/datum/game_mode/gang/post_setup()
+/datum/game_mode/gang/send_intercept()
 	set waitfor = FALSE
-	..()
-	sleep(rand(10,100))
+	..(FALSE, FALSE)
+
+	priority_announce("Ballas Federation Announcement", "SECURITY SHOULD OF SIDED WITH THE BALLAS!", 'sound/machines/Alarm.ogg')
```
Comment (ChangelingRain) [inline code/game/gamemodes/gang/gang.dm]: n o

**Comment by Steelpoint** [comment]
This seems like a very ham fisted way of removing security.

Also all I can see is this giving everyone on the station (who knows how the round will go) several minutes of free looting to prepare for when they likely get antag status.

I think the build up should be subtle at first and then explode into action to surprise most of the crew, not starting the action as soon as the round starts but not generating gangs until several minutes after everyone has made their own personal forts.

File: code/game/gamemodes/gang/gang.dm (hunk 11 above)
Comment (Iamgoofball) [inline code/game/gamemodes/gang/gang.dm]: Suggest better lore

Comment (Steelpoint) [inline code/game/gamemodes/gang/gang.dm]: Central Command Announcement: "WARNING! SECURITY HAS SUFFERED CATASTROPHIC DAMAGE., NO SURVIVORS. ENGINEERING PERSONNEL REPORT TO THE BRIG."

I would also suggest raising the station to Red Alert and declaring Martial Law.

**Comment by Cyberboss** [comment]
@Steelpoint this message comes 10 seconds before the explosion to give
greytiders and the captain time to evacuate

File: code/game/gamemodes/gang/gang.dm (hunk 11 above)
Comment (Iamgoofball) [inline code/game/gamemodes/gang/gang.dm]: You're on the wrong side of the street Grove motherfucker

**Comment by Robustin** [comment]
Still can't emphasize enough how much I hate the idea of delaying the round's progress so we can have 8 minutes of greytiding and metagaming. We hate extended, we see what happens when people KNOW its extended, I simply do not want to be in a mode that starts on that premise except instead of "theres no antags time to fuck around!" its "you're going to be an antag, REALLY fuck around!".

**Comment by Steelpoint** [comment]
Hence why I would be interested in a alternative solution, I think having three to eight minutes of a free for all with no security is going to negatively impact the round.

**Comment by Robustin** [comment]
I had ideas for this but its hard for me to touch them until this PR is resolved. My concept was to have "Vigilantes" replace security as the "loyalist" faction. They wouldn't be united, share comms, etc.  

The idea is that the vigilantes would be NT's (low cost) answer to the gang menace and any non-gangsters would start as vigilantes. Vigilantes can obviously still be converted since they're not implanted, but they would all have a device of their own that would process any gang contraband and return influence. That influence would work like gang influence and be used to buy weapons/equipment.

Currently the issue is:

1) No-sec Gang War starts
2) You see someone tagging your workplace
Without vigilantes:
3) You walk up and ask if they can convert you because fuck being neutral in a gang war without security
With vigilantes:
3) You push them down and sell their spray can for influence, then they come back later you use the switchblade you got from influence to shank them.

Vigilantes fix almost all the issues remaining with no-sec Gang War.

**Comment by Cyberboss** [comment]
@KorPhaeron, it was your idea

**Comment by Robustin** [comment]
Kor and I have been bouncing this back and forth. I brought up the notion of removing security from Gang War, he mentioned accomplishing that via an explosion (presumably as an IC way to take the armory out of play).

But this is one of the most crucial aspects of reforming GW and it has to be done right. A misstep here could easily leave the mode in worse shape than it was when it was put back in rotation.

I think this is a decent start, I'd use most of it for the PR I was going to do, but it has to address the elephant in the room which is "what do the non-gangsters do during a conversion mode with no security".

I really, really, really don't like the idea of just leaving them as-is (I know this playerbase and I know people would just basically start yelling on radio to recruit/be recruited. I don't see that as good gameplay and it raises a lot of balance issues when there's no "buffer" between the gangs, its just gangs and people who want to join gangs. I also don't like the idea of delays since its not just filler gameplay, its filler gameplay that will be incredibly toxic and metagamey. 

TL;DR: This shouldn't happen unless it includes Vigilantes (outlined above).

**Comment by KorPhaeron** [comment]
The eternal dilemnia of "nobody wants to review a non atomized PR" and "merging changes by themselves that are contingent on other changes will lead to a mess gameplay wise"

**Comment by Robustin** [comment]
It would be potentially manageable if this didn't have delays and didn't get merged until its sibling was merged. The Vigilante concept simply can't work with delays.

**Comment by Steelpoint** [comment]
I would prefer if sec still spawns but all/most sec equipment is removed from the station.

Lore it up that NT was in the process of relocating all Security personnel and equipment off station, but a sudden blockade from a Gang ship/space storm has prevented the remaining security personnel from leaving, and the gangs are now taking advantage of no security to attempt to ursurp control.

Maybe make it that if no gang has taken over the station by the 45 minute mark, a wave of security/ERT reinforcements arrive and the round ends with no one winning.

**Comment by Robustin** [comment]
@Steelpoint 

I think the problem with that is that sec is unified, has their own radio, "base", etc. and either you give them enough equipment to fight gangster, at which point they seize cargo at roundstart and its the same story or you don't give them gear and basically take a big shit on sec players for signing up.

**Comment by Steelpoint** [comment]
True.

At that point you either remove security from the game more entirely, or you only spawn one or two of them in, remove any way for them to recruit, and tell them to bunker down.

I dunno, we'll see how it goes.

**Comment by duncathan** [comment]
s t a l e

**Comment by Robustin** [comment]
I'll finish this

**Comment by Iamgoofball** [comment]
@robustin pls preserve the lore

**Comment by duncathan** [comment]
ok

**Comment by Steelpoint** [comment]
what lore?

**Comment by Iamgoofball** [comment]
>>>>>didnt read the PR
